addon/store.js

```diff
@@ -131,13 +123,41 @@ var Store = Ember.Object.extend({
 
             var attr = params[0];
             var value = options[attr];
-            var func = function(m) {
+
+            return this._findWithFilterFunc(type, function filterFunction(m) {
                 return get(m, attr) === value;
-            };
-            return this._findWithFilterFunc(type, func);
+            });
         }
+
         return this._findByIdComputed(type, options);
     },
+    findOne(type) {
+        var store = this;
+
+        return Ember.ObjectProxy.extend({
```

Comment:
I think it can be achieved without breaking functionality which is why I proposed it.  If you want to defer that's fine, I can land this on a branch for now since the app I am implementing this on will likely need it.